biofefi/utils/utils.py

```diff
@@ -125,4 +125,7 @@ def delete_directory(path: Path):
     Args:
         path: The path to the log directory
     """
-    shutil.rmtree(path, ignore_errors=True)
+    if path.is_file():
+        path.unlink()  # Remove the file
```

Comment:
I don't like the idea of including file behaviour in a function called delete_directory. I'd make another function for the file behaviour. This keeps the two semantically separate and it's clearer what each function does.